File: BlueMS/src/main/java/com/st/BlueMS/demos/ActivityRecognition/ActivityView.java
package com.st.BlueMS.demos.ActivityRecognition;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.constraint.ConstraintLayout;
import android.util.AttributeSet;
import android.widget.ImageView;

import com.st.BlueSTSDK.Features.FeatureActivity.ActivityType;


/**
 * common class between all the ones that display different activity icons
 */
abstract class ActivityView extends ConstraintLayout{

    /**
     * alpha to use for the selected image
     */
    private static final float SELECTED_ALPHA = 1.0f;

    /**
     * alpha to use for the other images
     */
    private static final float DEFAULT_ALPHA = 0.3f;

    /**
     * currently selected image
     */
    private ImageView mSelectedImage=null;

    public ActivityView(Context context) {
        super(context);
    }

    public ActivityView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ActivityView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    /**
     * utility method to deselect all the demo icons
     */
    protected void deselectAllImages(){
        for(ActivityType type : ActivityType.values()){
            ImageView image = getSelectedImage(type);
            if(image!=null)
                image.setAlpha(DEFAULT_ALPHA);
        }
    }

    /**
     * map an activity type to the linked imageview
     * @param status current activity
     * @return image view associated to the status activity or null if is an invalid status
     */
    abstract @Nullable
    ImageView getSelectedImage(@NonNull ActivityType status);


    /***
     * select a specific images
     * @param type activity image to select
     */
    public void setActivity(@Nullable ActivityType type) {
        //restore the alpha for the old activity
        if(mSelectedImage !=null)
            mSelectedImage.setAlpha(DEFAULT_ALPHA);

        //find the new activity
        if(type!=null)
            mSelectedImage = getSelectedImage(type);

        //set the alpha for the new activity
        if(mSelectedImage !=null)
            mSelectedImage.setAlpha(SELECTED_ALPHA);
    }

}

File: st_activity_recognition/src/main/java/com/st/activity_recognition/Activity/ActivityView.kt
/*
 * Copyright (c) 2022(-0001) STMicroelectronics.
 * All rights reserved.
 * This software is licensed under terms that can be found in the LICENSE file in
 * the root directory of this software component.
 * If no LICENSE file comes with this software, it is provided AS-IS.
 */
package com.st.activity_recognition.Activity

import android.content.Context
import android.util.AttributeSet
import android.widget.ImageView
import androidx.annotation.AttrRes
import androidx.constraintlayout.widget.ConstraintLayout
import com.st.blue_sdk.features.activity.ActivityType

/**
 * common class between all the ones that display different activity icons
 */
abstract class ActivityView : ConstraintLayout {
    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
    constructor(
        context: Context, attrs: AttributeSet?,
        @AttrRes defStyleAttr: Int
    ) : super(context, attrs, defStyleAttr)

    /**
     * currently selected image
     */
    private var mSelectedImage: ImageView? = null

    /**
     * utility method to deselect all the demo icons
     */
    protected fun deselectAllImages() {
        for (type in ActivityType.entries) {
            val image = getSelectedImage(type)
            if (image != null) {
                image.alpha = DEFAULT_ALPHA
            }
        }
    }

    /**
     * map an activity type to the linked imageview
     * @param status current activity
     * @return image view associated to the status activity or null if is an invalid status
     */
    abstract fun getSelectedImage(status: ActivityType): ImageView?

    /***
     * select a specific images
     * @param type activity image to select
     */
    fun setActivity(type: ActivityType?) {
        //restore the alpha for the old activity
        if (mSelectedImage != null) {
            mSelectedImage!!.alpha = DEFAULT_ALPHA
        }

        //find the new activity
        if (type != null) {
            mSelectedImage = getSelectedImage(type)
        }

        //set the alpha for the new activity
        if (mSelectedImage != null) {
            mSelectedImage!!.alpha = SELECTED_ALPHA
        }
    }

    companion object {
        /**
         * alpha to use for the selected image
         */
        private const val SELECTED_ALPHA = 1.0f

        /**
         * alpha to use for the other images
         */
        private const val DEFAULT_ALPHA = 0.3f
    }
}